### PR TITLE
Api 249 add check to ensure default namespace is not used

### PR DESCRIPTION
**Pull request by ilaflamme-va**
Is this a new product?
no

Has this been deployed successfully from a branch?
yes

Are load balancer rules being removed or renamed?
no

Are Kubernetes namespaces being removed or renamed?
no

Has DevOps been consulted on resource limits?
N/A

### CODE REVIEW

File: bin/validate-deployment-yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ $0 <FILE> <NAMESPACE> <WEAK_STRUCTURE_VALIDATION>
 
 ARGS:
   <FILE>
-    Input .yaml file to validate.
```
Comment (taylorrassmannlibertyits) [inline bin/validate-deployment-yaml:9]: I don't know if this is validating against default anymore? It looks like a lot of the code I wrote was taken out?

File: bin/validate-product-yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+EOF
+echo -e "$1" > /dev/stderr
+exit 1
+}
+
+FILE=$1
+
+[ -z $FILE ] && echo "No product.yaml specified." && usage && exit 1
+
+DEFAULT_NAMESPACE_TEST=$(grep -E '^.*namespace: *(default)?$' $FILE)
```
Comment (bschofield-va) [inline bin/validate-product-yaml:20]: Anyway we could use more descriptive names... that is, this isn't actually a test, it's illegal namespace declarations. Perhaps something `ILLEGAL_NAMESPACE_DECLARATIONS` or `BLACK_LISTED_NAMESPACES` is a more descriptive name for this value that provides a more clarity on _intent_.

```
[ -n "$ILLEGAL_NAMESPACE_DECLARATIONS" ] && fail "Empty or default namespaces used"
```
```

File: bin/validate-deployment-yaml (hunk 1 above)
Comment (ilaflamme-va) [reply bin/validate-deployment-yaml:9]: We're protected from default here by the clause you wrote:
`if [ "$DU_NAMESPACE" == "default" ]; then
  echo "Default namespace is not allowed. Please use a different namespace." && exit 1
fi`
We globally validate that the namespace is not  set to be "default".
In the _validate-deployment-yaml_ we validate that all of the namespaces provided here are equivalent to the `$DU_NAMESPAC` which was previously validated.

Comment (taylorrassmannlibertyits) [reply bin/validate-deployment-yaml:9]: I guess part of my concern is that from my understanding of the task, occasionally the namespace would still be default after substitution due to weird errors. That's why there was additional checking and validation going on.

**Pull request merged**